spec/helpers/hearing_paginator_spec.rb

```diff
@@ -0,0 +1,257 @@
+      end
+
+      context 'when current page is last page' do
+        before { instance.current_page = 3 }
+
+        it { is_expected.to be_falsey }
+      end
+    end
+  end
+
+  describe '#last_page' do
+    subject { instance.last_page }
+
+    context 'with no hearings' do
+      let(:hearings) { [] }
+
+      it { is_expected.to be(0) }
+    end
+
+    context 'with single hearing and hearing day' do
+      include_context 'with single hearing and hearing day'
+
+      it { is_expected.to be(0) }
+    end
+
+    context 'with multiple hearings and hearing days' do
+      include_context 'with multiple hearings and hearing days'
+
+      it { is_expected.to be(3) }
+    end
+  end
+
+  describe '#last_page?' do
+    subject { instance.last_page? }
+
+    context 'with single hearing and hearing day' do
+      include_context 'with single hearing and hearing day'
+
+      context 'when current page is last page' do
+        before { instance.current_page = 0 }
+
+        it { is_expected.to be_truthy }
+      end
+
+      context 'when current page is nil' do
+        before { instance.current_page = nil }
+
+        it { is_expected.to be_truthy }
+      end
+    end
+
+    context 'with multiple hearings and hearing days' do
+      include_context 'with multiple hearings and hearing days'
+
+      context 'when current page is first page' do
+        before { instance.current_page = 0 }
+
+        it { is_expected.to be_falsey }
+      end
+
+      context 'when current page is nil' do
+        before { instance.current_page = nil }
+
+        it { is_expected.to be_falsey }
+      end
+
+      context 'when current page is last page' do
+        before { instance.current_page = 3 }
+
+        it { is_expected.to be_truthy }
+      end
+    end
+  end
+
+  describe '#next_page_link' do
+    subject { instance.next_page_link }
+
+    include_context 'with multiple hearings and hearing days'
+
+    before { instance.current_page = instance.first_page }
+
+    it { is_expected.to have_link('Next hearing day') }
+
+    it {
+      is_expected.to have_link('Next hearing day',
+                               href: %r{/hearings/#{hearing1.id}\?page=1&urn=ACASEURN})
+    }
+
+    it {
+      is_expected.to have_link('Next hearing day',
+                               class: 'govuk-link app-pagination-next')
+    }
+  end
+
+  describe '#previous_page_link' do
+    subject { instance.previous_page_link }
+
+    before { instance.current_page = instance.last_page }
+
+    include_context 'with multiple hearings and hearing days'
+
+    it { is_expected.to have_link('Previous hearing day') }
+
+    it {
+      is_expected.to have_link('Previous hearing day',
+                               href: %r{/hearings/#{hearing2.id}\?page=2&urn=ACASEURN})
+    }
+
+    it {
+      is_expected.to have_link('Previous hearing day',
+                               class: 'govuk-link app-pagination-previous')
+    }
+  end
```

Comment:
Should this spec be extended to include tests for describe '#hearing_page_items' and describe '#sorted_hearing_page_items'.

Reply:
Those are private methods and therefore should not be tested directly.

To [quote others](https://anthonysciamanna.com/2016/02/14/should-private-methods-be-tested.html)
>> The short answer is that you shouldn’t test private methods directly, but only their effects on the public methods that call them. 

They are tested via the public `#items` method, which as you can see calls `#sorted_hearing_page_items`, which in turn calls `#hearing_page_item`

Reply:
👍